src/test/java/core/basesyntax/strategy/AddOperationHandlerTest.java

```diff
@@ -0,0 +1,134 @@
+package core.basesyntax.strategy;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.dto.Transaction;
+import core.basesyntax.model.Fruit;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class AddOperationHandlerTest {
+    private static final String PURCHASE = "p";
+    private static final String SUPPLY = "s";
+    private static final String BALANCE = "b";
+    private static final String RETURN = "r";
+    private static final Fruit APPLE = new Fruit("apple");
+    private static final Fruit BANANA = new Fruit("banana");
+    private static final Fruit ORANGE = new Fruit("orange");
+    private static Map<String, OperationHandler> operationStrategyMap;
+
+    @BeforeClass
+    public static void before() {
+        operationStrategyMap = new HashMap<>();
+        operationStrategyMap.put(PURCHASE, new PurchaseOperationHandler());
+        operationStrategyMap.put(SUPPLY, new AddOperationHandler());
+        operationStrategyMap.put(BALANCE, new BalanceOperationHandler());
+        operationStrategyMap.put(RETURN, new AddOperationHandler());
+    }
+
+    @Before
+    public void beforeEach() {
+        Storage.storage.clear();
+        Storage.storage.put(APPLE, 20);
+        Storage.storage.put(BANANA, 30);
+        Storage.storage.put(ORANGE, 40);
+    }
+
+    @Test
+    public void supplySingle() {
+        int expected = 50;
+        Transaction transaction = new Transaction(SUPPLY, APPLE, 30);
+        operationStrategyMap.get(transaction.getOperation()).apply(transaction);
```

Comment:
do we need this map and calling `get()`?
we can create only the handler instance that we are testing